src/hyp3_sdk/hyp3.py

```diff
@@ -46,6 +46,13 @@ def __init__(self, api_url: str = PROD_API, username: Optional[str] = None, pass
 
         self.session = hyp3_sdk.util.get_authenticated_session(username, password)
         self.session.headers.update({'User-Agent': f'{hyp3_sdk.__name__}/{hyp3_sdk.__version__}'})
+        self._check_application_status()
+
+    def _check_application_status(self) -> None:
+        info = self.my_info()
+        if info['application_status'] != 'APPROVED':
+            warnings.warn(f'User {info["user_id"]} has not yet applied for a monthly credit allotment.'
+                          ' Please visit https://hyp3.asf.alaska.edu/request_access to submit your application.')
```

Comment:
Will https://hyp3.asf.alaska.edu/request_access be the correct link?

Reply:
We're aiming for https://hyp3-docs.asf.alaska.edu/using/requesting_access

Reply:
The warning message(s) can match the error message(s) we implemented in HyP3 at https://github.com/ASFHyP3/hyp3/blob/develop/lib/dynamo/dynamo/exceptions.py#L19 for not started/pending/rejected

Reply:
@asjohnston-asf so would you suggest different warning messages for no submitted application, pending application, and rejected?

Reply:
If you like. I documented them as stretch goals in the comments for TOOL-2740, but I'm not going to insist if you don't think it's worth the effort.